cmd/podman/main.go

```diff
@@ -36,6 +38,21 @@ func main() {
 		return
 	}
 
+	shell := os.Getenv("SHELL")
+	if filepath.Base(os.Args[0]) == "podmansh" || strings.HasSuffix(os.Args[0], filepath.Base(shell)) {
+		podmanshCommand := &cobra.Command{}
```

Comment:
I haven't verified if it works but if the whole goal is just to mimic `podman exec -it podmansh` then maybe we can just trick us into going into that code path by overwriting `os.Args`, i.e. just set
```
os.Args = []string{"podman", "exec", "-it", "podmansh", "/bin/sh"}
```
Then cobra will just parse that and we will go into the right exec code path without adding any new code.

Reply:
Clever idea, worth an experiment to see if it works.

Reply:
One issue with this could be that we can not retry if the container is not ready yet.

Reply:
Shouldn't you manage that via the systemd dependencies instead, i.e. make sure the container unit is up and running before any login can be made?

Reply:
Although it looks like we might have a bug

```
$ podman container inspect --format '{{ .State.Running }}' podmansh
true
$ podman container wait --condition=running podmansh
-1
```

Reply:
I guess this is just the way that podman container wait works, it prints out the exit code, since the container has not exited, it is printing -1.  The podman container wait returns nil in this case.

Reply:
Well podman wait does not support it either, with --sdnotify conmon which is the default in quadlet it will wait for the same time as `podman container wait --condition=running` would. At that point we know that the container pid1 is running which means all following `podman exec`'s should just work.

Reply:
But if there is no way to block the execution of podmansh untile after the quadlet is started, then we have to wait in podmansh.